### ISSUE TITLE
Support all MDX components exported by Gatsby Theme Carbon

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon-platform#322]
We need to support MDX components provided in the Gatsby theme: 

https://github.com/carbon-design-system/gatsby-theme-carbon/blob/main/packages/gatsby-theme-carbon/index.js

**SM**

- AnchorLinks, AnchorLink MDX components
- PageDescription MDX component
- Caption MDX component
- Accordion, AccordionItem MDX components
- InlineNotification MDX component
- Title MDX component
- Aside MDX component

Components below take in images as children and are blocked by #499.

**MD**

- Video MDX component
- GifPlayer MDX component
- DoDontRow, DoDont MDX components
- ArtDirection MDX component

**LG**

- Tabs, Tab MDX components
- ResourceCard (resource-card-group styles) MDX component
- ArticleCard MDX component
- MediumPosts MDX component

**Not in scope for v1**

- FeatureCard
- ImageCard
- SquareCard
- ImageGallery,  ImageGalleryImage
- ExpressiveListContainer
- ExpressiveList
- DoDontExample (deprecated)
